src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java

```diff
@@ -186,6 +266,50 @@
  *   &lt;property name=&quot;setterCanReturnItsClass&quot; value=&quot;true&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <pre>
+ * public class SomeClass {
+ *
+ *   private String field;
+ *   private String testField;
+ *
+ *   public SomeClass(String testField) { // violation, 'testField' param hides 'testField' field
+ *   }
+ *   public void method(String param) { // OK
+ *       String field = param; // violation, 'field' variable hides 'field' field
+ *   }
+ *   public void setTestField(String testField) { // OK, 'testField' param doesn't hide any field
+ *       this.field = field;
+ *   }
+ *   public SomeClass setField(String field) { // OK, 'field' param doesn't hide any field
+ *       this.field = field;
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check so that it ignores parameters of abstract methods:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;HiddenField&quot;&gt;
+ *   &lt;property name=&quot;ignoreAbstractMethods&quot; value=&quot;true&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <pre>
+ * abstract class SomeClass {
+ *
+ *   private String field;
+ *
+ *   public SomeClass(int field) { // violation, 'field' param hides a 'field' field
+ *     float field; // violation, 'field' variable hides a 'field' field
+ *   }
+ *   public abstract int method(String field); // OK
+ * }
+ * public class Demo extends SomeClass {
```

Comment:
please add empty line above this line to ease reading.

Reply:
done